Sources/Target/Linux/X86_64/ProcessX86_64.cpp

```diff
@@ -34,7 +55,8 @@ ErrorCode Process::allocateMemory(size_t size, uint32_t protection,
   }
 
   // MAP_FAILED is -1.
-  if (static_cast<int64_t>(result) == -1LL) {
+  if ((is32 && static_cast<int64_t>(result) == -1LL) ||
+    (!is32 && static_cast<int32_t>(result) == -1)) {
```

Comment:
Mhmmm yeah, I think that's right. Let me fix it.


On Tue, May 10, 2016 at 2:55 AM, kokarez <notifications@github.com> wrote:

> In Sources/Target/Linux/X86_64/ProcessX86_64.cpp
> <https://github.com/facebook/ds2/pull/335#discussion_r62642923>:
>
> > @@ -34,7 +55,8 @@ ErrorCode Process::allocateMemory(size_t size, uint32_t protection,
> >    }
> >
> >    // MAP_FAILED is -1.
> > -  if (static_cast<int64_t>(result) == -1LL) {
> > +  if ((is32 && static_cast<int64_t>(result) == -1LL) ||
> > +    (!is32 && static_cast<int32_t>(result) == -1)) {
>
> I think this doesn't match LLVM coding style
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/335/files/2b1edf99ba9bf9243133bf874f6098b42f6b25c8..40b3c02af90a28d5cd1551520ec168071c1af0dd#r62642923>
>



-- 
Stephane Sezer